google/cloud/storage/internal/grpc_object_request_parser.cc

```diff
@@ -364,6 +364,59 @@ GrpcObjectRequestParser::ToProto(PatchObjectRequest const& request) {
   return result;
 }
 
+StatusOr<google::storage::v2::UpdateObjectRequest>
+GrpcObjectRequestParser::ToProto(UpdateObjectRequest const& request) {
+  google::storage::v2::UpdateObjectRequest result;
+  auto status = SetCommonObjectParameters(result, request);
+  if (!status.ok()) return status;
+  SetGenerationConditions(result, request);
+  SetMetagenerationConditions(result, request);
+  SetCommonParameters(result, request);
+  SetPredefinedAcl(result, request);
+
+  auto& object = *result.mutable_object();
+  object.set_bucket("projects/_/buckets/" + request.bucket_name());
+  object.set_name(request.object_name());
+  object.set_generation(request.GetOption<Generation>().value_or(0));
+
+  result.mutable_update_mask()->add_paths("acl");
+  for (auto const& a : request.metadata().acl()) {
+    *object.add_acl() = GrpcObjectAccessControlParser::ToProto(a);
+  }
+
+  // The semantics in gRPC are to replace any metadata attributes
+  result.mutable_update_mask()->add_paths("metadata");
+  for (auto const& kv : request.metadata().metadata()) {
```

Comment:
seems like metadata is optional: https://cloud.google.com/storage/docs/json_api/v1/objects/update#request-body

should we update the mask if `request.metadata().metadata()` is empty?

Reply:
That would have the effect of leaving the metadata unchanged, which is certainly not what we want. I am not sure there is a way to express "unset" vs. "empty" metadata in gRPC.

Reply:
And I am not too worried either, both have the same semantics.